docs/paper/dev/event-api/custom-events.md

```diff
@@ -0,0 +1,122 @@
+---
+slug: /dev/event-api/custom-events
+---
+
+# Custom Events
+
+Creating custom events is a great way to add functionality to your plugin. 
+This will allow for other people to listen for your custom events and add functionality to your plugin.
+
+## Creating a custom event
+
+To create a custom event, you need to create a class that extends `Event`. Each event has a `HandlerList` that contains all the listeners that are listening for that event.
+
+This list is used to call the listeners when the event is called.
+    
+```java title="PaperIsCoolEvent.java"
+
```

Comment:
I'd get rid of the newlines at the beginning of each codeblock. It isn't what is done elsewhere and adds weird looking extra space. Let me know if there's a specific reason you have these though